modules/core/src/test/java/org/apache/ignite/internal/TestRecordingCommunicationSpi.java

```diff
@@ -72,6 +72,27 @@
     /** */
     private volatile IgniteBiInClosure<ClusterNode, Message> c;
 
+    /**
+     * When the filed is true an message will be unwrapped before it will be passed to a predicate.
+     */
+    private final boolean unwrapOriginMessage;
+
+    /**
+     * Default constructor.
+     * Messages are unwrapped before them are passed to a predicate by default.
+     */
+    public TestRecordingCommunicationSpi() {
+        this(true);
+    }
+
+    /**
+     * @param unwrapOriginMessage True when a message will be unwrapped before provide to a predicate,
+     * false the message will be provided as an Io message.
+     */
+    public TestRecordingCommunicationSpi(boolean unwrapOriginMessage) {
```

Comment:
I think that we do not need to modify `TestRecordingCommunicationSpi` itself. Using a custom `Spi` is more straightforward I think. Something like this:
`
    public static class MyCustomSpi extends TestRecordingCommunicationSpi {
        /** Security subject id. */
        private volatile UUID subjectId;

        /** {@inheritDoc} */
        @Override public void sendMessage(
            ClusterNode node,
            Message msg,
            IgniteInClosure<IgniteException> ackC
        ) throws IgniteSpiException {
            if (msg instanceof GridIoSecurityAwareMessage && subjectId != null) {
                GridIoSecurityAwareMessage ioSecMsg = (GridIoSecurityAwareMessage)msg;

                log.info("Msg is sending to node [msg=" + ioSecMsg.message().getClass().getSimpleName() +
                    ", node=" + node.consistentId() + ']');

                GridTestUtils.setFieldValue(msg, GridIoSecurityAwareMessage.class, "secSubjId", subjectId);
            }

            super.sendMessage(node, msg, ackC);
        }

        public void securitySubjectId(UUID subjectId) {
            this.subjectId = subjectId;
        }
    }
`

Reply:
Fixed.